lib/rucio/client/richclient.py

```diff
@@ -309,16 +310,14 @@ def get_pager() -> 'Callable[[str], None]':
     :returns: pager
     """
     default_pager = 'less'
-    try:
-        result = subprocess.run([default_pager], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
-        less_available = result.returncode == 0
-    except Exception:
-        less_available = False
+
+    default_pager_available = shutil.which(default_pager) is not None
+
+    if not default_pager_available:
+        return pydoc.pager
 
     def less_pager_function(text: str) -> None:
         """Use the 'less' pager with -FRSXKM options."""
         pydoc.pipepager(text, f'{default_pager} -FRSXKM')
 
-    if less_available:
-        return less_pager_function
-    return pydoc.pager
+    return less_pager_function
```

Comment:
I'd personally rewrite it like this for clarity:

```python
    # if `less` is available, set default options
    if shutil.which('less') is not None:
        return lambda text: pydoc.pipepager(text, 'less -FRSXKM')
    # fall back to available pager
    else:
        return pydoc.pager
```

Reply:
Agree, refactored!